src/utilities/device_utils.c

```diff
@@ -130,6 +130,17 @@ hypre_DeviceDataDestroy(hypre_DeviceData *data)
    }
 #endif // #if defined(HYPRE_USING_CUSPARSE) || defined(HYPRE_USING_ROCSPARSE)
 
+#if defined(HYPRE_USING_CUSOLVER) || defined(HYPRE_USING_ROCSOLVER)
+   if (data->vendor_solver_handle)
+   {
+#if defined(HYPRE_USING_CUSOLVER)
+      HYPRE_CUSOLVER_CALL(cusolverDnDestroy(data->vendor_solver_handle));
+#else
+      HYPRE_ROCBLAS_CALL(rocblas_destroy_handle(data->vendor_solver_handle));
```

Comment:
I am not familiar with rocsolver yet.. so you want call `rocblas_*` here (and at other places) instead of `rocsolver_*` if there exist such things... it just looks strange.

Reply:
rocSOLVER is the ROCm analog to cuSOLVER. Think "LAPACK". But rocSOLVER will internally use rocblas and, therefore, some rocSOLVER APIs want a handle to rocBLAS.

Reply:
Thanks @pbauman! Feel free to chime-in this PR if you have any coding suggestions. Thanks!

Reply:
Thanks @pbauman for clarifying.